include/RevThread.h

```diff
@@ -0,0 +1,109 @@
+//
+// _RevThread_h_
+//
+// Copyright (C) 2017-2023 Tactical Computing Laboratories, LLC
+// All Rights Reserved
+// contact@tactcomplabs.com
+//
+// See LICENSE in the top level directory for licensing details
+//
+//
+
+#ifndef _SST_REVCPU_REVTHREAD_H_
+#define _SST_REVCPU_REVTHREAD_H_
+
+#define __INVALID_TID__ 0x0
+
+// -- Standard Headers
+#include <cstdint>
+#include <vector>
+#include <set>
+
+// -- Rev Headers
+#include "RevMem.h"
+#include "RevInstTable.h"
+
+using namespace SST::RevCPU;
+using MemSegment = RevMem::MemSegment;
+
+// Enum for tracking the state of a RevThread.
+enum class ThreadState {
+    START,   // Initial state; resources are allocated.
+    RUNNING, // Thread is assigned and executing on a Proc/HART.
+    BLOCKED, // Waiting for I/O or thread synchronization.
+    READY,   // In the ready list; waiting for CPU (not implemented yet).
+    DONE     // Thread has finished; deallocate resources.
+};
+
+class RevThread {
+public:
+    RevThread(uint32_t inputThreadID, uint32_t inputParentThreadID,
+              uint64_t inputStackPtr, uint64_t inputFirstPC,
+              std::shared_ptr<MemSegment>& inputThreadMem,
+              RevFeature* inputFeature);
+
+    // ThreadID operations
+    uint32_t GetThreadID(){ return ThreadID; }
```

Comment:
Getter functions should be `const` if possible (if not, consider a `mutable` member).